src/core/eval.ml

```diff
@@ -217,8 +218,16 @@ and tree_walk : dtree -> ctxt -> stack -> (term * stack) option =
         in
         List.iter fn env_builder;
         (* Complete the array with fresh meta-variables if needed. *)
+        let make_t =
+          begin
+            match meta_type with
+            | Some meta_type -> (function () -> make_meta [] meta_type)
+            | None -> (function () -> let x = make_meta [] Type in
```

Comment:
why not puting () outside: `let make_t () = ....` ?
```
let make_t () =
  match meta_type with
    (* FIXME? why using [] and not ctxt ? *)
            | Some meta_type -> make_meta [] meta_type
            | None -> make_meta [] (make_meta [] Type)
in
```
@gabrielhdt what do you think of using ctxt instead of []?

Reply:
this part has been removed... but the question of the context : ctxt vs []  has remains relevant